applications/connectivity_bridge/src/disk/config.c

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <fs/fs.h>
+#include <settings/settings.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#define MODULE file_config
+#include "module_state_event.h"
+#include "fs_event.h"
+#include "ble_ctrl_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, LOG_LEVEL_DBG);
+
+#if CONFIG_FS_FATFS_LFN
+#define FILE_NAME         "Config.txt"
+#else
+#define FILE_NAME         "CONFIG.TXT"
+#endif
+
+#define SETTINGS_KEY "bridgecfg"
+
+#define SETTINGS_KEY_BLE "ble"
+#define SETTINGS_NAME_BLE "BLE_ENABLED"
+
+static const char file_contents_header[] =
+"==========================================\r\n"
+"  Nordic Thingy:91 Configuration options\r\n"
+"==========================================\r\n"
+"The parameters below can be changed at runtime.\r\n"
+"\r\n"
+"NOTE: For changes to take effect,\r\n"
+"safely disconnect (unmount) the drive and disconnect the USB cable.\r\n"
+"==========================================\r\n";
+
+static bool ble_enabled;
+
+static void send_ble_ctrl_event(bool enable)
+{
+	struct ble_ctrl_event *event = new_ble_ctrl_event();
+
+	event->cmd = enable ? BLE_CTRL_ENABLE : BLE_CTRL_DISABLE;
+	EVENT_SUBMIT(event);
+}
+
+static void save_settings(void)
+{
+	int err;
+
+	err = settings_save_one(
+			SETTINGS_NAME_BLE, &ble_enabled, sizeof(ble_enabled));
+	if (err) {
+		LOG_ERR("settings_save_one: %d", err);
+	}
+}
+
+static int settings_set(const char *key, size_t len_rd,
+			settings_read_cb read_cb, void *cb_arg)
+{
+	if (strcmp(key, SETTINGS_KEY_BLE) == 0) {
+		u32_t val;
```

Comment:
Let's initialize val so that if read_cb accidentally doesn't set val, then at least we aren't de-referencing uninitialzed memory.